common/ts/prepare-task.ts

```diff
@@ -118,6 +122,12 @@ function branchName(fullName: string) {
   return fullName;
 }
 
+export function reportPath(): string {
+  return `${tl.getVariable('Agent.TempDirectory')}\\${tl.getVariable(
```

Comment:
I'm pretty sure the temp directory is deleted by the agent after each build so creating a sub-directory using the build ID is probably unnecessary. However, I can't find any documentation to confirm this and it's better to be safe.

It might be worth creating a `sonar` subdirectory to help with documentation/diagnosis e.g. if a user is trying to diagnose a build failure on their agent they probably won't know what is create those subfolders or why. However, if they saw the files were under a directory called `sonar` they'd be able to work it out.

i.e. make the root temp directory something like this:  {Agent.TempDirectory}\sonar\{BuildId}\

Reply:
The reason of using guid is just to handle multiple report-task.txt generated during a build (it's already supported for maven/gradle builds), so as long as we generate the path in the prepare task, and we don't even know the project key nor some kind of unique identifier of the currently built project, that's the pretty much only solution to achieve that. But i agree for the sonar directory, i can add it to the generated path.

Reply:
Absolutely, we need the `guid` subdir; I didn't mean to suggest we didn't. Sorry for not being clearer.

Reply:
Oh, ok, no problem, sorry too.